doc/posts/2023/2023_06_12_vara_week3.rst

```diff
@@ -0,0 +1,21 @@
+VQ-VAE results and study on Diffusion models : Week 3
+=====================================================
+
+.. post:: June 12, 2023
+     :author: Vara Lakshmi Bayanagari
+     :tags: google
+     :category: gsoc
+
+What I did this week
+~~~~~~~~~~~~~~~~~~~~
+I continued my experiments with VQ-VAE on MNIST data to see the efficacy of Prior training in the generated outputs. The output of encoder for every input image delivers a categorical index of a latent vector for every pixel in the output. As discussed in the previous blog post, prior has been trained separately using PixelCNN (without any conditioning) in the latent space. If PixelCNN is a bunch of convolutions, then what makes it a generative model? This is an important question to ask and the answer to it is the sampling layer used on pixelCNN outputs during inference. The official code in Keras uses a tfp.layers.DistributionLambda(tfp.distributions.Categorical) layer as its sampling layer. Without this sampling layer PixelCNN outputs are deterministic and collapse to single output. Also similarly, sampling layer alone, i.e., without any PixelCNN trained prior, on the pre-determined outputs of encoder is deterministic. This is due to the fact that latent distances are correctly estimated by the pre-trained encoder and during inference categorical sampling layer would always sample the least distance latent, i.e., the one closest to the input. Therefore, the autoregressive nature of PixelCNN combined with a sampling layer for every pixel delivers an effective generative model. The outputs for all my experiments are shown in the image below -
```

Comment:
- `Prior`:  why uppercase ?
- `previous blog post`: please add a link (check how to link to rst file)
-  `tfp.layers.DistributionLambda(tfp.distributions.Categorical)`.  : missing code block quotes

Reply:
Prior is in capitals since it's a proper noun here, otherwise it may lead to confusions in understanding the term correctly. Updated the blog with other comments.